src/test/java/LoginTests.java

```diff
@@ -1,25 +1,19 @@
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.chrome.ChromeDriver;
-import org.openqa.selenium.chrome.ChromeOptions;
+
 import org.testng.Assert;
 import org.testng.annotations.Test;
-
-import java.time.Duration;
+import pages.HomePage;
+import pages.LoginPage;
 
 public class LoginTests extends BaseTest {
-    @Test
-    public void LoginEmptyEmailPasswordTest() {
 
-//      Added ChromeOptions argument below to fix websocket error
-        ChromeOptions options = new ChromeOptions();
-        options.addArguments("--remote-allow-origins=*");
+    @Test
+    public void loginValidCredentialsTest() {
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
 
-        WebDriver driver = new ChromeDriver(options);
-        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        loginPage.login();
+        homePage.waitForOverlay(overlayLocator);
```

Comment:
The code is not compiled! Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/f2ee2a95-bcc2-440f-b817-6890c06c39e0)

HINT

**Step 1**

**public class BasePage**

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/ee7642ae-4e29-45bb-86ac-b703285d9c21)

    public  void waitForOverlayElement (WebElement element) {
        wait.until(ExpectedConditions.visibilityOf(element));
    }
This is a Java method that uses Selenium Webdriver's ExpectedConditions class to wait until a certain WebElement becomes visible on the webpage. The wait object used is an instance of WebDriverWait, which is initialized with a timeout value. The visibilityOf method of the ExpectedConditions class waits until the specified element becomes visible on the page, at which point the wait.until() method returns control to the caller. This method is commonly used to ensure that a webpage has finished loading an element or overlay before continuing with further actions.

**Step 2**

**public class HomePage**

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/a49f7c79-bd29-44f6-a48c-6a5dbce2aa29)

 Java method that returns a WebElement representing an avatar or profile picture icon on a webpage. The method uses the waitForOverlayElement method, which we've previously discussed, to ensure that the avatar icon is visible on the page before returning it.
 
 **Step 3**
 
 ![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/94ec9635-bfb4-4ce5-995b-8b9ea199b59b)
 
 
![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/662eb5e9-193b-43f9-854e-8acbd79a8b62)